pkg/client.go

```diff
@@ -16,11 +16,17 @@
 // implementations.
 package pkg
 
-// SecretClient provides a contract for retrieving secrets from a secret store manager.
+// SecretClient provides a contract for storing and retrieving secrets from a secret store provider.
 type SecretClient interface {
 	// GetSecrets retrieves secrets from a secret store.
```

Comment:
It is protected by the ACL policies to the authenticated vault tokens in the vault, and the authToken is set on the request header

Reply:
> Is the assumption that the path leads to only leaf nodes, that is keys?

No.  The path in the secret store is built of the the Path in configuration.toml plus the path parameter passed to the method.  So let's say you had a username and password for two cloud providers for your uploaddata service.

configuration.toml
[ SecretService ]
Path = /v1/secret/edgex/uploaddata

Then you could call
GetSecrets("provider1", []string{"username","password")
GetSecrets("provider2", []string{"username","password")

However, this is equivalent to

configuration.toml
[ SecretService ]
Path = /v1/secret/edgex

Then you could call
GetSecrets("uploaddata/provider1", []string{"username","password")
GetSecrets("uploaddata/provider2", []string{"username","password")

Moreover, there is nothing restricting you from giving your secret store hierarchy and storing secrets at every level

/v1/secret/edgex/uploaddata
   versionsecret
   upstreams
       provider1
       provider2
   downstreams
       provider3
       provider4

However, in reality, there is only this in the secret store.

/v1/secret/edgex/uploaddata/versionsecret
/v1/secret/edgex/uploaddata/upstreams/provider1
/v1/secret/edgex/uploaddata/upstreams/provider2
/v1/secret/edgex/uploaddata/downstreams/provider3
/v1/secret/edgex/uploaddata/downstreams/provider4

Your default access token limits to you /v1/secret/edgex/uploaddata subtrees.

> Assuming service level path check using auth token is carried out, so Core-Data service token cannot access App Function SDK secrets when each service has different tokens. 

Correct.  However, this can be overridden by configuration at time of token creation.  Also note that no access policies are enforced today due to the use of the root token.

>But would this protect App Function SDK accessible secrets from different tenants.

I've designed multitenant systems before and EdgeX has no concept of a tenant.

> For example, say we are going to export data to an infrastructure management endpoint and to a domain endpoint, one using AWS and another Google cloud.

An attacker that compromises the Edge may be able to access both providers if they get to uploaddata's secrets.  Whether provider1 can access provider2 or vice-versa is unknowable and dependent upon the design of the upstream service.

> In Security WG meeting @lenny-intel and @bnevis-i indicated safe. But capturing concern here to be sure.